src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -61,42 +65,96 @@ public async Task<bool> PushMetadataAsync()
                 }
                 else
                 {
-                    List<BuildData> buildsManifestMetadata = GetBuildManifestsMetadata(ManifestsPath);
+                    List<BuildData> buildsManifestMetadata = GetBuildManifestsMetadata(ManifestsPath, cancellationToken);
 
                     BuildData finalBuild = MergeBuildManifests(buildsManifestMetadata);
 
                     IMaestroApi client = ApiFactory.GetAuthenticated(MaestroApiEndpoint, BuildAssetRegistryToken);
-                    Client.Models.Build recordedBuild = await client.Builds.CreateAsync(finalBuild, s_cancellationToken);
+
+                    var deps = await GetBuildDependenciesAsync(client, cancellationToken);
+                    Log.LogMessage(MessageImportance.High, "Calculated Dependencies:");
+                    foreach (var dep in deps)
+                    {
+                        Log.LogMessage(MessageImportance.High, $"    {dep.BuildId}, IsProduct: {dep.IsProduct}");
+                    }
+                    finalBuild.Dependencies = await GetBuildDependenciesAsync(client, cancellationToken);
+
+                    Client.Models.Build recordedBuild = await client.Builds.CreateAsync(finalBuild, cancellationToken);
 
                     Log.LogMessage(MessageImportance.High, $"Metadata has been pushed. Build id in the Build Asset Registry is '{recordedBuild.Id}'");
                 }
             }
             catch (Exception exc)
             {
-                Log.LogErrorFromException(exc, true);
+                Log.LogErrorFromException(exc, true, true, null);
             }
 
             return !Log.HasLoggedErrors;
         }
 
+        private async Task<IImmutableList<BuildRef>> GetBuildDependenciesAsync(
+            IMaestroApi client,
+            CancellationToken cancellationToken)
+        {
+            var logger = new MSBuildLogger(Log);
+            var local = new Local(LocalHelpers.GetGitDir(logger), logger);
+            IEnumerable<DependencyDetail> dependencies = await local.GetDependenciesAsync();
+            var builds = new Dictionary<int, bool>();
+            foreach (var dep in dependencies)
+            {
+                var buildId = await GetBuildId(dep, client, cancellationToken);
```

Comment:
Hrm. I wonder if we should just push the asset list during the "new build" and let the server do this work. That prevents a lot of back and forth chatter with the API... to tell translate things it can translate.  Unless there is some sort of local data we can't expressin "list of used dependencies" or something?

Reply:
This could probably be calculated on the server, but I'd rather do this logic on the build machine where we can more easily get logs and see if something is wrong.